File: Java/Uva11044.java
import java.util.Scanner;

public class Main {
    public static void main(String args[]){
        Scanner sc = new Scanner(System.in);
        int t;
        t = sc.nextInt();
        while(t-- != 0){
            int n,m;
            n = sc.nextInt();
            m = sc.nextInt();
            double ans = Math.ceil(((double)(n-2)/3))* Math.ceil(((double)(m-2)/3));
            System.out.println((int)ans);
        }
    }
}

File: C++/UVa11044.cpp
#include<iostream>
#include<math.h>

using namespace std;

int main(){
    int t;
    cin >> t;
    while(t--){
        int n,m;
        cin >> n >> m;
        int ans = ceil((double(n-2))/3)*ceil((double(m-2))/3);
        cout << ans << endl;

    }
    return 0;
}
